test/langtools/tools/javac/launcher/SourceLauncherTest.java

```diff
@@ -598,6 +598,38 @@ public void testClassInModule(Path base) throws IOException {
                 "error: compilation failed");
     }
 
+    @Test
+    public void testNoRecompileWithSuggestions(Path base) throws IOException {
+        tb.writeJavaFiles(base,
+            "class NoRecompile {\n" +
+            "    void use(String s) {}\n" +
+            "    void test() {\n" +
+            "        use(1);\n" +
+            "    }\n" +
+            "    <T> void test(T t, Object o) {\n" +
+            "        T t1 = (T) o;\n" +
+            "    }\n" +
+            "    static class Generic<T> {\n" +
+            "        T t;\n" +
+            "        void raw(Generic raw) {\n" +
+            "            raw.t = \"\";\n" +
+            "        }\n" +
+            "    }\n" +
+            "    void deprecation() {\n" +
+            "        Thread.currentThread().stop();\n" +
+            "    }\n" +
+            "    void preview(Object o) {\n" +
+            "      if (o instanceof String s) {\n" +
+            "          System.out.println(s);\n" +
+            "      }\n" +
```

Comment:
Is there a reason you're not using a text block here?

Reply:
Using text block will prevent potential "clean" backports of this issue. However if you think this patch won't be subject of backport, I can turn it into text block.

Reply:
👍